packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -818,6 +1143,137 @@ contract StargateAdapter is Ownable, IBridgeAdapter, ILayerZeroComposer {
         );
     }
 
+    /**
+     * @dev Try to deposit assets to FleetCommander without reverting on failure
+     */
+    function _tryDepositToFleetCommander(
+        address fleetCommander,
+        address asset,
+        uint256 amount,
+        address shareRecipient,
+        bytes memory referralCode,
+        bytes32 operationId,
+        address, // originalUser - not used in current implementation
+        address, // sourceAsset - not used in current implementation
+        uint16 sourceChainId
+    ) internal returns (bool success) {
+        // Validate FleetCommander supports the asset
+        try IFleetCommanderMinimal(fleetCommander).asset() returns (
+            address fleetAsset
+        ) {
+            if (fleetAsset != asset) {
+                emit CrossChainFleetDepositFailed(
```

Comment:
not sure it's the best way to handle the failures - the users will be burdened with the full gas fee. Unless that reverts elswhere

Reply:
now I see it's on the destination chain. Then I believe the reverts could be caught on top level `_handle..()` methods - and sent back to the owner on failure ( on destination chain ) - not sure where LZ / Stargate will send it back if it's a unhandeled revert.

Reply:
I've added some additional code to handle the failure such that if the deposit on the target chain fails we refund the user on that chain.